echelon_form.g

```diff
@@ -0,0 +1,31 @@
+# This file creates a matrix in strict row echelon form to be used for testing and a function that gets a matrix and returns another matrix with the same strict row echelon form.
+
+echelonMat := function(height, width, rank, randomSeed, ring)
+   local echelonMat, rightCorner, i, j;
+
+    rightCorner := RandomMat(randomSeed, rank, width-rank, ring);
+    echelonMat := NullMat(height, width, ring);
+
+    for i in [1 .. rank] do
+        echelonMat[i][i] := One(ring);
+    od;
+    echelonMat := echelonMat;
+
+    for i in [1 .. rank] do
+        for j in [1 .. (width-rank)] do
+            echelonMat[i][rank+j] := rightCorner[i][j];
+        od;
+    od;
+
+    return echelonMat;
+end;
+
+shapelessMat := function(mat, height, width, randomSeed, ring)
+    local i, grp, left;
+
+    grp := GL(height, ring);
+    i := Length(GeneratorsOfGroup(grp));
+    Group_InitPseudoRandom(grp, i+3, 20);
```

Comment:
Can you add a short source comment on why we do this?

Reply:
Done.